src/components/graph/search/GraphSearch.tsx

```diff
@@ -109,10 +111,18 @@ function GraphSearch(props: Props) {
 				e.keyCode === KeyCodes.ENTER
 			) {
 				onTimestampSubmit(currentTimestamp);
+				if (!refreshPanels) {
+					setShowModal(false);
+					return;
+				}
+				if (timestamp) {
```

Comment:
there something wrong with state in this component
`const [timestamp, setTimestamp] = React.useState<number | null>(null);` might be redundant. I'll look it up later

We should rely on inputConfig.timestamp. and we got into this block it's already valid timestamp no need to check it.
These if statemenets make hard to understand what is going on here.

I guess it should be
```
if (refreshPanels) {
	setMode('confirm')
} else {
	setShowModal(false);
}
```